design-proposals/klip-34-optional-with-in-c-star-stmts.md

```diff
@@ -0,0 +1,220 @@
+);
+```
+
+Switch these properties to _optional_ would allow more natural SQL syntax:
+
+```sql
+CREATE TABLE FOO (
+   ID BIGINT PRIMARY KEY,
+   VAL STRING
+);
+```
+
+This is more intuitive and familiar to new users with a SQL background.
+
+## Why required `KAFKA_TOPIC`?
+
+It's very common to see `CREATE` statements where the `KAFKA_TOPIC` name matches the name of the stream
+or table being created, maybe with different case.
+
+`KAFKA_TOPIC` is already optional for `CREATE STREAM AS SELECT` and `CREATE TABLE AS SELECT` statements.
+
+There seems no reason to _require_ users to supply the topic name where it matches the name of the 
+created stream or table.
+
+## Why required `VALUE_FORMAT`?
+
+Many companies standardize on a single format, e.g. they're an 'Avro House' or a 'JSON house'. It seems
+strange to require each and every `CREATE STREAM/TABLE` statement to have to spell out the value format.
+
+`VALUE_FORMAT` is already optional for `CREATE STREAM AS SELECT` and `CREATE TABLE AS SELECT` statements.
+
+[KLIP-33](https://github.com/confluentinc/ksql/pull/6017) is introducing a `KEY_FORMAT` to mirror the
+`VALUE_FORMAT`. This will not be a required value. Instead, a system configuration 
+`ksql.persistence.default.format.key` can be set to provide the default key format where no explicit
+`KEY_FORMAT` is supplied.
+
+Like `KEY_FORMAT`, users should be able to choose a default value format, rather than having to specify
+it in every statement. If they're a 'JSON house' they can just set the default value format to `JSON`
+and never worry about it again.  Alternatively, users could set the default in their session properties.
+
+## What is in scope
+
+* Addition of new server configuration to provide a default for value format.
+* Removal of requirement for `VALUE_FORMAT` property in the `WITH` clause of CT/CS statements, where
+  the server configuration provides a default.
+* Removal of requirement for `KAFKA_FORMAT` property in the `WITH` clause of CT/CS statements.
+
+## What is not in scope
+
+Anything and everything else =)
+
+## Value/Return
+
+Clearer and cleaner syntax, with less unnecessary typing. Lowers barrier of entry for new users who
+are already familiar with Ansi SQL.
+
+## Public APIS
+
+### Server configs
+
+The following new configuration options will be added. These configurations can be set globally, 
+within the application property file, or locally, via the `SET` command.
+
+* `ksql.persistence.default.format.value`: the default value format. Default: none.
+
+* `ksql.persistence.topic.name.case`: if `lower`, defaults ksql to using lowercase topics names
+  where topic names are not explicitly provided. This includes matching a `CREATE TABLE` or 
+  `CREATE STREAM` statement to an existing topic, and includes the case of sink topics created
+  by `CREATE STREAM AS SELECT` and `CREATE TABLE AS SELECT`. Default: `upper`.
+
+* `ksql.persistence.topic.name.case.insensitive`: if `true`, matching a `CREATE TABLE` or 
```

Comment:
Same comment as above. This config doesn't feel necessary if we preserve case between stream/table names and topic names.

Reply:
As above, this is currently possible, but clunky